higher/optim.py

```diff
@@ -139,6 +140,7 @@ def step(
         loss: _torch.Tensor,
         params: _typing.Iterable[_torch.Tensor] = None,
         override: _typing.Optional[_OverrideType] = None,
+        preproc_grad: _typing.Callable[[list], list] = None,
```

Comment:
I started writing this functionality in a private branch before the release. Feel free to adapt the docstring I used as necessary:

```
grad_callback: (optional) a single argument function which will be
                applied to a list of gradients of parameters, which respects the
                order specified by `reference_params`. This can be used to apply
                a function, such as gradient clipping, to all (or a subset) of
                these gradients every time the step function is called. This
                callback overrides the default provided when constructing the
                differentiable optimizer.
```

Please note that I also allowed people to specify the `grad_callback` at optimizer construction time, with the following docstring:
```
grad_callback: (optional) a single argument function which will be
                applied to a list of gradients of parameters, which respects the
                order specified by `reference_params`. This can be used to apply
                a function, such as gradient clipping, to all (or a subset) of
                these gradients every time the step function is called. If this
                keyword argument is provided when calling the step method, its
                value will override the default specified here.
```